drivers/sensor/lm77/lm77.c

```diff
@@ -0,0 +1,425 @@
+	case SENSOR_ATTR_UPPER_THRESH:
+		reg = LM77_REG_THIGH;
+		break;
+	case SENSOR_ATTR_ALERT:
+		reg = LM77_REG_TCRIT;
+		break;
+	case SENSOR_ATTR_HYSTERESIS:
+		reg = LM77_REG_THYST;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	lm77_sensor_value_to_temp(val, &temp);
+
+	err = lm77_write_temp(dev, reg, temp);
+	if (err) {
+		LOG_ERR("failed to write register 0x%02x (err %d)", reg, err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int lm77_attr_get(const struct device *dev, enum sensor_channel chan,
+			 enum sensor_attribute attr, struct sensor_value *val)
+{
+	int16_t temp;
+	uint8_t reg;
+	int err;
+
+	__ASSERT_NO_MSG(val != NULL);
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_LOWER_THRESH:
+		reg = LM77_REG_TLOW;
+		break;
+	case SENSOR_ATTR_UPPER_THRESH:
+		reg = LM77_REG_THIGH;
+		break;
+	case SENSOR_ATTR_ALERT:
+		reg = LM77_REG_TCRIT;
+		break;
+	case SENSOR_ATTR_HYSTERESIS:
+		reg = LM77_REG_THYST;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	err = lm77_read_temp(dev, reg, &temp);
+	if (err) {
+		LOG_ERR("failed to read register 0x%02x (err %d)", reg, err);
+		return err;
+	}
+
+	lm77_temp_to_sensor_value(temp, val);
+
+	return 0;
+}
+
+#if LM77_TRIGGER_SUPPORT
+static int lm77_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+			    sensor_trigger_handler_t handler)
+{
+	const struct lm77_config *config = dev->config;
+	struct lm77_data *data = dev->data;
+	gpio_flags_t flags;
+	int err;
+
+	__ASSERT_NO_MSG(trig != NULL);
+
+	if (trig->type != SENSOR_TRIG_THRESHOLD || trig->chan != SENSOR_CHAN_AMBIENT_TEMP ||
+	    config->int_gpio.port == NULL) {
+		return -ENOTSUP;
+	}
+
+	if (handler) {
+		flags = GPIO_INT_EDGE_TO_ACTIVE;
+	} else {
+		flags = GPIO_INT_DISABLE;
+	}
+
+	err = gpio_pin_interrupt_configure_dt(&config->int_gpio, flags);
+	if (err) {
+		LOG_ERR("failed to configure INT GPIO IRQ (err %d)", err);
+		return err;
+	}
+
+	data->trigger = trig;
+	data->trigger_handler = handler;
+
+	return 0;
+}
+
+static void lm77_trigger_work_handler(struct k_work *item)
+{
+	struct lm77_data *data = CONTAINER_OF(item, struct lm77_data, work);
+	sensor_trigger_handler_t handler = data->trigger_handler;
+
+	if (handler) {
+		handler(data->dev, (struct sensor_trigger *)data->trigger);
+	}
+}
+
+static void lm77_int_gpio_callback_handler(const struct device *port,
+					   struct gpio_callback *cb,
+					   gpio_port_pins_t pins)
+{
+	struct lm77_data *data = CONTAINER_OF(cb, struct lm77_data, int_gpio_cb);
+
+	ARG_UNUSED(port);
+	ARG_UNUSED(pins);
+
+	k_work_submit_to_queue(&data->workq, &data->work);
+}
+#endif /* LM77_TRIGGER_SUPPORT */
+
+static int lm77_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct lm77_data *data = dev->data;
+	int16_t temp;
+	int err;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	err = lm77_read_temp(dev, LM77_REG_TEMP, &temp);
+	if (err) {
```

Comment:
nit: better use an explicit boolean expression, e.g. `err < 0` (ref. MISRA 14.4)

Reply:
@gmarull Thanks. I have changed all `if` expressions to be _essentially boolean_ type.